bad_code.java

```diff
@@ -0,0 +1,54 @@
+public class BadCodeExample {
+    
+    public static void main(String[] args) {
+        ChildClass child = new ChildClass();
+        child.doSomething();
+        child.doSomethingElse();
+    }
+}
+
+class ParentClass {
+    public int someNumber = 10;
+    public String someString = "Hello";
+
+    public void doSomething() {
+        System.out.println("Doing something in ParentClass");
+        someNumber++;
+        someString += " Worlds";
```

Comment:
Consider using a more descriptive string manipulation in `ParentClass`'s `doSomething` method. For example, use `String.format` for better readability and efficiency. [medium]